### ISSUE TITLE
Sort feature bug

### ISSUE THREAD

**Issue by GavinVM** [GavinVM/mr-tracker#20]
When trying to reorder films via the re-order feature they move ok but when you go to save the screen twitches for a lack of a better description and the icon to save stays in place rather than saving the change.

Console log
`mrTracker.TrackerPage.handleReorder:: starting
mrTracker.TrackerPage.handleReorder:: finishing
mrTracker.TrackerPage.handleReorder:: starting
mrTracker.TrackerPage.handleReorder:: finishing
mrTracker.TrackerPage.handleReorder:: starting
mrTracker.TrackerPage.handleReorder:: finishing
mrTracker.TrackerPage.toggleReorder:: starting
mrTracker.TrackerPage.toggleReorder:: setting to false
mrTracker.TrackerPage.toggleReorder:: saving changes
mrTracker.StorageService.setEntry:: starting
mrTracker.StorageService.setEntry:: passed in parameters key: trackerList, value:  (6) [{…}, {…}, {…}, {…}, {…}, {…}]
mrTracker.TrackerPage.toggleReorder:: finishing
mrTracker.StorageService.setEntry:: item saved for key trackerList
mrTracker.StorageService.setEntry:: finishing
N {__zone_symbol__state: null, __zone_symbol__value: Array(0)}__zone_symbol__state: true__zone_symbol__value: (6) [{…}, {…}, {…}, {…}, {…}, {…}]Symbol(Symbol.species): (...)Symbol(Symbol.toStringTag): (...)[[Prototype]]: Object
mrTracker.TrackerPage.trackerList:: starting
mrTracker.TrackerPage.trackerList:: trackerlist reset
mrTracker.TrackerPage.trackerList:: finishing`

**Comment by GavinVM** [comment]
after looking into the code I found that there both an issue with setting up the storage item for the list and when the reorder is saved it wasn't resetting the ui properly.

The storage was set up with only having one item initially but when the cex list was added and local storage was used to store that later the creating a fresh list was failings as there was no step to handle a storage setup with more than one item that wasnt the tracking list. Once this was handled both adding and re-order became more stable, then few locations that utilised the previous tracking list flow were updated. Once this was refactored, the reorder toggle flow was updated to create a toast on success as well as failure and revert the ui in both cases, leaving the list alone in the case of success.

Once this was checked test case updated to reflect new flows.
